### ISSUE TITLE
[TD-50] synchronizing gets out of sync(?)

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#51]
_Ben Boeckel on 2014-04-29T05:49:56Z says:_ 

 I have 2 clients which are syncing with a server. This morning, I got an error in Mirakel about an unknown error message. Investigating this evening, I see the following in the logs:

    ERROR: Could not find common ancestor for bc1ec10c-5af0-46fb-b7da-162d6f7bb2d4

This task is on both clients. First, more debugging information would be nice (what did the server try? how can it be fixed?). Second, it'd be nice to have instructions on the wiki for getting out of ruts.

While trying to get things back to a sensible state, I rolled back to a backup of the server data, but it seems that there is shared state in the clients as well and they think they know where the server is supposed to be and only send what they think is new (this may be a Mirakel bug, but I *think* I'm seeing something similar in task), so there's a gap in the data now. When trying to delete all of the data and re-sync from scratch (which Mirakel doesn't seem to be able to do), things still seem weird. Guidance?

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-04-29T05:49:56Z
Modified: 2015-02-07T19:51:09Z
```

**Comment by taskwarrior** [comment]
_Ben Boeckel on 2014-04-29T05:57:58Z says:_ 

 Mirakel bug: https://github.com/MirakelX/mirakel-android/issues/372

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-08-13T14:02:35Z says:_ 

 Hi Ben, 

This seems to be a Mirakel related problem and therefore not something we can fix.

As for the second part "shared state in the clients", recovering the server data may not be enough to fix a problem with syncing. If any of the clients already has incorrect or divergent data the problem will simply reveal itself again.

If you cannot attach the server data as well as the client data from all the clients you sync, we won't know what went wrong.
There have been reports of "getting out of sync" but they turned out to be caused by external actions like manual tinkering with data files.
Basically any action on the data files that is not performed by the taskwarrior client is outside our control.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2015-02-07T19:51:09Z says:_ 

 Looks like a known Mirakel bug, and no further complaints made.
